hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -1809,6 +1809,10 @@ public boolean balance(boolean force) throws IOException {
 
       List<RegionPlan> plans = this.balancer.balanceCluster(assignments);
 
+      if (runMode.isDryRun()) {
+        return true;
+      }
+
```

Comment:
@joshelser thanks for the review! I wanted to point out that a few lines below this diff (line 1825) is the `postBalance` coprocessor call. We skip that for dry run mode, but a few lines above this diff (line 1790) is the `preBalance` coprocessor call, which we _do_ call.

I chose to call pre and not post because it seems like maybe someone might use `preBalance` to check or set extra state that might be necessary for the balancer, while I was concerned that `postBalance` might make decisions based on balance plans that are passed in but were not actually executed.

I've been thinking maybe we should just skip both or call both, and wondering if you have any thoughts. We could also pass in the RunMode into the coprocessor, but then we get into coprocessor interface changes. Not sure that's worth it?

Reply:
I followed the same logic in the rsgroup implementation

Reply:
FYI I pushed a commit which disables the preBalance coprocessor for dry run mode so we are consistent. Open to other approaches.